### ISSUE TITLE
Deps file for MSBuild shouldn't downgrade common task dependencies.

### ISSUE THREAD

**Issue by ericstj** [dotnet/cli#11444]
## Steps to reproduce
Produce a task with a dependency on a new version of roslyn.

## Expected  behavior
Task works on latest SDK

## Actual behavior
Task fails to load with ref-def mismatch for roslyn DLLs since old version (2.1.0) is listed in MSBuild's deps file and carried in the MSBuild bin folder.

## Environment data
`dotnet --info` output:
```
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview6-011681
 Commit:    2f5626d019

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   D:\dotnet\corefx\.dotnet\sdk\3.0.100-preview6-011681\

Host (useful for support):
  Version: 3.0.0-preview6-27629-07
  Commit:  a3967b6096

.NET Core SDKs installed:
  2.2.103 [D:\dotnet\corefx\.dotnet\sdk]
  3.0.100-preview6-011681 [D:\dotnet\corefx\.dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.2.1 [D:\dotnet\corefx\.dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.1 [D:\dotnet\corefx\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview6-19230-05 [D:\dotnet\corefx\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.1 [D:\dotnet\corefx\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview6-27629-07 [D:\dotnet\corefx\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview6-27629-07 [D:\dotnet\corefx\.dotnet\shared\Microsoft.WindowsDesktop.App]
```

@nguerrera suggested a tactical fix https://github.com/dotnet/arcade/issues/2748#issuecomment-497140703

**Comment by nguerrera** [comment]
Draft PR here: https://github.com/dotnet/toolset/pull/1127

Waiting for the change to remove dotnet-migrate, which will just eliminate roslyn altogether from this root dir and deps.

**Comment by nguerrera** [comment]
There is still one more fx file in root dir, but not in deps that is not up-to-date: System.Xml.XPath.XmlDocument. It comes in via insertion from vstest. I will file a bug on them. They should retarget past netcoreapp2.1 from netcoreapp1.0 and this should go away.

**Comment by nguerrera** [comment]
https://github.com/microsoft/vstest/issues/2043

**Comment by nguerrera** [comment]
Migrate is gone: #11481 🎉

**Comment by nguerrera** [comment]
Reopening to wait for a build with both #11481 and  dotnet/toolset#1127, and I will validate that the everything is lifted up or removed.

**Comment by nguerrera** [comment]
Confirmed fixed in 3.0.100-preview7-012271.
